contracts/bonding/BondingManager.sol

```diff
@@ -332,68 +335,141 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
+                // If caller is a registered transcoder and is no longer bonded, resign
+                resignTranscoder(msg.sender);
+            }
+        } 
+
+        Unbond(del.delegateAddress, msg.sender, unbondingLockId, _amount, withdrawRound);
+    }
 
-        // Delegator no longer bonded to anyone
-        del.delegateAddress = address(0);
-        // Unbonding delegator does not have a start round
-        del.startRound = 0;
+    /**
+     * @dev Rebond tokens for an unbonding lock to a delegator's current delegate while a delegator
+     * is in the Bonded or Pending states
+     * @param _unbondingLockId ID of unbonding lock to rebond with
+     */
+    function rebond(
+        uint256 _unbondingLockId
+    ) 
+        external
+        whenSystemNotPaused
+        currentRoundInitialized 
+        autoClaimEarnings
+    {
+        // Caller must not be an unbonded delegator
+        require(delegatorStatus(msg.sender) != DelegatorStatus.Unbonded);
 
-        Unbond(del.delegateAddress, msg.sender);
+        // Process rebond using unbonding lock
+        processRebond(msg.sender, _unbondingLockId);
     }
 
     /**
-     * @dev Withdraws bonded stake to the caller after unbonding period.
+     * @dev Rebond tokens for an unbonding lock to a delegate while a delegator
+     * is in the Unbonded state
+     * @param _to Address of delegate
+     * @param _unbondingLockId ID of unbonding lock to rebond with
      */
-    function withdrawStake()
+    function rebondFromUnbonded(
+        address _to,
+        uint256 _unbondingLockId
+    )
         external
         whenSystemNotPaused
         currentRoundInitialized
+        autoClaimEarnings
     {
-        // Delegator must be in the unbonded state
+        // Caller must be an unbonded delegator
         require(delegatorStatus(msg.sender) == DelegatorStatus.Unbonded);
 
-        uint256 amount = delegators[msg.sender].bondedAmount;
-        delegators[msg.sender].bondedAmount = 0;
-        delegators[msg.sender].withdrawRound = 0;
+        // Set delegator's start round and transition into Pending state
+        delegators[msg.sender].startRound = roundsManager().currentRound().add(1);
+        // Set delegator's delegate
+        delegators[msg.sender].delegateAddress = _to;
+        // Process rebond using unbonding lock
+        processRebond(msg.sender, _unbondingLockId);
+    }
+
+    /**
+     * @dev Withdraws tokens for an unbonding lock that has existed through an unbonding period
+     * @param _unbondingLockId ID of unbonding lock to withdraw with
+     */
+    function withdrawStake(uint256 _unbondingLockId)
+        external
+        whenSystemNotPaused
+        currentRoundInitialized
+    {
+        Delegator storage del = delegators[msg.sender];
+        UnbondingLock storage lock = del.unbondingLocks[_unbondingLockId];
+
+        // Unbonding lock must be valid
+        require(lock.withdrawRound > 0);
```

Comment:
I see two instances of checking if the lock.withdrawRound > 0 determining whether the unbonding lock is valid? Should we move to a convenience method for code readability and reducing the chance of a future error when trying to figure out what it means for a lock to be valid?